src/pkcs11.cpp

```diff
@@ -502,7 +524,8 @@ int pkcs11_remove_keys(lmp_options &opt)
 	if (PKCS11_login(slot, 0, opt.hsm_pin.c_str()))
 		leave;
 
-	remove_keys(slot->token, hsm_cfg.tls_lbl);
+	remove_lmp_keys(slot->token);
```

Comment:
we can check the return value and leave on error

Reply:
I was about to  comment on that, since I intentionally reverted  your change. Since that happens inside the cleanup routine, I believe is better to just keep removing what we can, and not skip PKCS11_logout if an error happens here. I can keep your changes if you do not agree with that.

Reply:
I think it is better to be consistent on how we handle errors than doing our best and then get random results to analyze. This way we know exactly where we are failing and why...so I'd rather fail early really.

I sent you another patch with some more stuff.

Reply:
Inside `populate_sota_dir` we either need a new macro instead of `leave_exit`, or we need to keep `int ret` defined. I'm re-adding `int ret` for now to proceed with testing.

Reply:
Based on your other changes I'm replacing `leave_exit` with `goto error` in `populate_sota_dir` to fix the build.

Reply:
Applied your changes + the required fixes to allow code to compile.